From cb992cb5b3099fdb4106d558f1aeba908267a075 Mon Sep 17 00:00:00 2001
From: Marek Kasik <mkasik@redhat.com>
Date: Tue, 9 Apr 2024 11:01:01 +0200
Subject: [PATCH] clipboard: Do not crash during disconnection

Disconnect handler of "owner-change" in finalize of
FrdpChannelClipboard so that the handler is not called
after the clipboard is finalized.

Fixes #50
---
 src/frdp-channel-clipboard.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/frdp-channel-clipboard.c b/src/frdp-channel-clipboard.c
index 2723c2d..ef82dbe 100644
--- a/subprojects/gtk-frdp/src/frdp-channel-clipboard.c
+++ b/subprojects/gtk-frdp/src/frdp-channel-clipboard.c
@@ -203,6 +203,9 @@ frdp_channel_clipboard_finalize (GObject *object)
   FrdpChannelClipboard        *self = (FrdpChannelClipboard *) object;
   FrdpChannelClipboardPrivate *priv = frdp_channel_clipboard_get_instance_private (self);
 
+  g_signal_handler_disconnect (priv->gtk_clipboard,
+                               priv->clipboard_owner_changed_id);
+
   g_hash_table_unref (priv->remote_files_requests);
   fuse_session_unmount (priv->fuse_session);
   fuse_session_exit (priv->fuse_session);
-- 
2.44.0

